### PR TITLE
[Bug]Solve the jobid is empty cause canceled inaccurate Patch

### PR DESCRIPTION
**Pull request by MonsterChenzhuo**
Problem Summary:

### What is changed and how it works?

Proposal: [xxx](url) 

Feel free to ping committers for the review!

## Contribution Checklist

  - Make sure that the pull request corresponds to a [GITHUB issue](https://github.com/apache/streampark/issues).

  - Name the pull request in the form "[Feature] [component] Title of the pull request", where *Feature* can be replaced by `Hotfix`, `Bug`, etc.

  - Minor fixes should be named following this pattern: `[hotfix] [docs] Fix typo in README.md doc`.

### CODE REVIEW

**Comment by wolfboys** [comment]
I'll review later.

**Comment by wolfboys** [comment]
@lvshaokang @xinzhuxiansheng PTAL

File: streampark-flink/streampark-flink-submit/streampark-flink-submit-core/src/main/scala/org/apache/streampark/flink/submit/impl/KubernetesNativeApplicationSubmit.scala [hunk 1]
```diff
@@ -71,7 +71,7 @@ object KubernetesNativeApplicationSubmit extends KubernetesNativeSubmitTrait {
         .getClusterClient
 
       val clusterId = clusterClient.getClusterId
-      val result = SubmitResponse(clusterId, flinkConfig.toMap)
+      val result = SubmitResponse(clusterId, flinkConfig.toMap, submitRequest.flinkJobID)
```
Comment (xinzhuxiansheng) [inline streampark-flink/streampark-flink-submit/streampark-flink-submit-core/src/main/scala/org/apache/streampark/flink/submit/impl/KubernetesNativeApplicationSubmit.scala:74]: `submitResponse` suggests that you don't need to set flinkJobId again. When you start the job, setting jobId, which can be strictly followed. There is only one set. If there are no special circumstances, it is recommended not to have a set, which can reduce the burden of code reading.

submitResponse建议不用再设置 flinkJobId, 当在启动Job时候开始设置，那可以严格的遵守，只有1个set，若无特殊情况，建议不要有set，这样可以减少代码阅读负担。 下面代码 submitResponse可以不用在setJobId

`ApplicationServiceImpl`
```java
if (StringUtils.isNoneEmpty(submitResponse.jobId())) {
    application.setJobId(submitResponse.jobId());
}
```

Comment (xinzhuxiansheng) [inline streampark-flink/streampark-flink-submit/streampark-flink-submit-core/src/main/scala/org/apache/streampark/flink/submit/impl/KubernetesNativeApplicationSubmit.scala:74]: `submitResponse` suggests that you don't need to set flinkJobId again. When you start the job, setting jobId, which can be strictly followed. There is only one set. If there are no special circumstances, it is recommended not to have a set, which can reduce the burden of code reading.

submitResponse建议不用再设置 flinkJobId, 当在启动Job时候开始设置，那可以严格的遵守，只有1个set，若无特殊情况，建议不要有set，这样可以减少代码阅读负担。 下面代码 submitResponse可以不用在setJobId

`ApplicationServiceImpl`
```java
if (StringUtils.isNoneEmpty(submitResponse.jobId())) {
    application.setJobId(submitResponse.jobId());
}
```

**Review by lvshaokang — DISMISSED** [review]
@MonsterChenzhuo LGTM. Thanks!

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/ApplicationServiceImpl.java [hunk 2]
```diff
@@ -1188,6 +1187,7 @@ public void start(Application appParam, boolean auto) throws Exception {
 
         assert application != null;
 
+        application.setJobId(new JobID().toHexString());
```
Comment (youyangkou) [inline streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/ApplicationServiceImpl.java:1190]: If the jobid is generated actively, then the copy method in ApplicationServiceImpl class also needs to add the code of creating new jobid, for example: 
```
newApp.setJobId (new JobID (). ToHexString ())
```

**Review by xinzhuxiansheng — APPROVED** [review]
LGTM

**Review by youyangkou — CHANGES_REQUESTED** [review]
Please optimization the copy method in ApplicationServiceImpl class.

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/ApplicationServiceImpl.java (hunk 2 above)
Comment (xinzhuxiansheng) [reply streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/ApplicationServiceImpl.java:1190]: setJobId is only executed at start job and does not affect the copy logic

Comment (xinzhuxiansheng) [reply streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/ApplicationServiceImpl.java:1190]: > If the jobid is generated actively, then the copy method in ApplicationServiceImpl class also needs to add the code of creating new jobid, for example:
> 
> ```
> newApp.setJobId (new JobID (). ToHexString ())
> ```

@youyangkou  setJobId is only executed at start job and does not affect the copy logic

**Review by wolfboys — APPROVED** [review]
LGTM

**Pull request merged**